photon-ml/src/integTest/scala/com/linkedin/photon/ml/GameTestUtils.scala

```diff
@@ -0,0 +1,247 @@
+      GLMOptimizationConfiguration(), treeAggregateDepth, isTrackingState)
+
+  /**
+    * Generates a fixed effect model
+    *
+    * @param featureShardId the feature shard id of the model
+    * @param dimensions the number of dimensions
+    * @return the newly generated fixed effect model
+    */
+  def generateFixedEffectModel(featureShardId: String, dimensions: Int) = new FixedEffectModel(
+    sc.broadcast(LogisticRegressionOptimizationProblem.initializeZeroModel(dimensions)),
+    featureShardId)
+
+  /**
+    * Generates a fixed effect coordinate and model
+    *
+    * @param featureShardId the feature shard id of the model
+    * @param size the size of the dataset
+    * @param dimensions the number of dimensions
+    * @param seed the random seed
+    * @return problem coordinate and random effect model
+    */
+  def generateFixedEffectCoordinateAndModel(
+      featureShardId: String,
+      size: Int,
+      dimensions: Int,
+      seed: Int = DefaultSeed) = {
+
+    val dataset = generateFixedEffectDataSet(featureShardId, size, dimensions, seed)
+    val optimizationProblem = generateFixedEffectOptimizationProblem()
+    val coordinate = new FixedEffectCoordinate[LogisticRegressionModel, TwiceDiffFunction[LabeledPoint]](
+      dataset, optimizationProblem)
+
+    val model = generateFixedEffectModel(featureShardId, dimensions)
+
+    (coordinate, model)
+  }
+
+  /**
+    * Generates a random effect dataset
+    *
+    * @param individualIds a set of random effect individual ids
+    * @param randomEffectId the random effect id
+    * @param featureShardId the feature shard id
+    * @param size the size of the dataset
+    * @param dimensions the number of dimensions
+    * @param seed the random seed
+    * @param numPartitions the number of spark partitions
+    * @return the newly generated random effect dataset
+    */
+  def generateRandomEffectDataSet(
+      individualIds: Seq[String],
+      randomEffectId: String,
+      featureShardId: String,
+      size: Int,
+      dimensions: Int,
+      seed: Int = DefaultSeed,
+      numPartitions: Int = 4) = {
+
+    val datasets = individualIds.map((_,
+      new LocalDataSet(
+        generateLabeledPoints(size, dimensions, seed)
+          .map(addGlobalId)
+          .toArray)))
+
+    val partitioner = new HashPartitioner(numPartitions)
+    val globalIdToIndividualIds = sc.parallelize(
+      individualIds.map(addGlobalId)).partitionBy(partitioner)
+    val activeData = sc.parallelize(datasets).partitionBy(partitioner)
+
+    new RandomEffectDataSet(
+      activeData, globalIdToIndividualIds, None, None, randomEffectId, featureShardId)
+  }
+
+  /**
+    * Generates random effect models
+    *
+    * @param individualIds a set of random effect individual ids for which to generate models
+    * @param dimensions the number of dimensions
+    * @return the newly generated random effect model
+    */
+  def generateRandomEffectModels(
```

Comment:
This name is a little misleading, because unlike the generateFixedEffectModel method which generates a FixedEffectModel object, this method generates some GeneralizedLinearModels used elsewhere to create a single RandomEffectModel